src/crypto/crypto.cpp

```diff
@@ -8,6 +8,58 @@
 /***********************************************************************************************************************
  * Public
  **********************************************************************************************************************/
+aos::Error Crypto::Init()
+{
+    auto ret = psa_crypto_init();
+
+    return ret != PSA_SUCCESS ? AOS_ERROR_WRAP(ret) : aos::ErrorEnum::eNone;
+}
+
+aos::Error Crypto::CreateCSR(
+    const aos::crypto::x509::CSR& templ, const aos::crypto::PrivateKeyItf& privKey, aos::Array<uint8_t>& pemCSR)
+{
+    mbedtls_x509write_csr csr;
+    mbedtls_pk_context    key;
+
+    InitializeCSR(csr, key);
+
+    auto cleanup = [&]() {
+        mbedtls_x509write_csr_free(&csr);
+        mbedtls_pk_free(&key);
+    };
+
+    mMutex.Lock();
```

Comment:
If you guard Add/Remove key, use static mutex in driver itself.

Reply:
Done